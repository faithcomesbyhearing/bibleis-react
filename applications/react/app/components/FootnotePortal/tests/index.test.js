import React from 'react';
import { mount } from 'enzyme';
import FootnotePortal from '..';

// Mocks
Object.defineProperty(document, 'getElementById', {
	value: (input) => {
		const el = document.createElement('div');
		el.id = input;
		return el;
	},
});

const props = {
	message: 'Please Login.',
	coords: { x: 15, y: 15 },
	closeFootnote: jest.fn(),
};

describe('FootnotePortal component', () => {
	it('should match snapshot with expected props', () => {
		const tree = mount(<FootnotePortal {...props} />);
		expect(tree).toMatchSnapshot();
	});
});
