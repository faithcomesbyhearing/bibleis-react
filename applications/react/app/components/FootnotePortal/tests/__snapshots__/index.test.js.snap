// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FootnotePortal component should match snapshot with expected props 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <FootnotePortal
    closeFootnote={[MockFunction]}
    coords={
      Object {
        "x": 15,
        "y": 15,
      }
    }
    message="Please Login."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "closeFootnote": [MockFunction],
      "coords": Object {
        "x": 15,
        "y": 15,
      },
      "message": "Please Login.",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "portal",
      "props": Object {
        "children": <div
          className="footnote-portal-container"
          style={
            Object {
              "left": 15,
              "top": 15,
            }
          }
        >
          <span
            onClick={[MockFunction]}
            role="button"
            tabIndex={0}
          >
            X
          </span>
          <p>
            Please Login.
          </p>
        </div>,
        "containerInfo": <div
          id="__next"
        >
          <div
            class="footnote-portal-container"
            style="left: 15px; top: 15px;"
          >
            <span
              role="button"
              tabindex="0"
            >
              X
            </span>
            <p>
              Please Login.
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="footnote-portal-container"
          style="left: 15px; top: 15px;"
        >
          <span
            role="button"
            tabindex="0"
          >
            X
          </span>
          <p>
            Please Login.
          </p>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              onClick={[MockFunction]}
              role="button"
              tabIndex={0}
            >
              X
            </span>,
            <p>
              Please Login.
            </p>,
          ],
          "className": "footnote-portal-container",
          "style": Object {
            "left": 15,
            "top": 15,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <span
              role="button"
              tabindex="0"
            >
              X
            </span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "X",
              "onClick": [MockFunction],
              "role": "button",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              "X",
            ],
            "type": "span",
          },
          Object {
            "instance": <p>
              Please Login.
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please Login.",
            },
            "ref": null,
            "rendered": Array [
              "Please Login.",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.portal),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "closeFootnote": [MockFunction],
        "coords": Object {
          "x": 15,
          "y": 15,
        },
        "message": "Please Login.",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "portal",
        "props": Object {
          "children": <div
            className="footnote-portal-container"
            style={
              Object {
                "left": 15,
                "top": 15,
              }
            }
          >
            <span
              onClick={[MockFunction]}
              role="button"
              tabIndex={0}
            >
              X
            </span>
            <p>
              Please Login.
            </p>
          </div>,
          "containerInfo": <div
            id="__next"
          >
            <div
              class="footnote-portal-container"
              style="left: 15px; top: 15px;"
            >
              <span
                role="button"
                tabindex="0"
              >
                X
              </span>
              <p>
                Please Login.
              </p>
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="footnote-portal-container"
            style="left: 15px; top: 15px;"
          >
            <span
              role="button"
              tabindex="0"
            >
              X
            </span>
            <p>
              Please Login.
            </p>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                onClick={[MockFunction]}
                role="button"
                tabIndex={0}
              >
                X
              </span>,
              <p>
                Please Login.
              </p>,
            ],
            "className": "footnote-portal-container",
            "style": Object {
              "left": 15,
              "top": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <span
                role="button"
                tabindex="0"
              >
                X
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "X",
                "onClick": [MockFunction],
                "role": "button",
                "tabIndex": 0,
              },
              "ref": null,
              "rendered": Array [
                "X",
              ],
              "type": "span",
            },
            Object {
              "instance": <p>
                Please Login.
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please Login.",
              },
              "ref": null,
              "rendered": Array [
                "Please Login.",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": Symbol(react.portal),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
